### PR TITLE
ref(similarity): Rollout similarity delete record to 20% EA

### PR DESCRIPTION
**Pull request by jangjodi**
Project option `sentry:similarity_backfill_completed` is `True` for all EA projects
Rollout delete record by hash feature to 20% of these projects

### CODE REVIEW

File: src/sentry/tasks/delete_seer_grouping_records.py [hunk 1]
```diff
@@ -13,6 +14,7 @@
 from sentry.tasks.base import instrumented_task
 
 logger = logging.getLogger(__name__)
+EA_ROLLOUT_PERCENTAGE = 20
```
Comment (mrduncan) [inline src/sentry/tasks/delete_seer_grouping_records.py:17]: Can we use an option here?

Comment (jangjodi) [reply src/sentry/tasks/delete_seer_grouping_records.py]: good catch, thanks!

File: src/sentry/tasks/delete_seer_grouping_records.py [hunk 2]
```diff
@@ -56,7 +57,13 @@ def call_delete_seer_grouping_records_by_hash(
         project = group.project if group else None
     if (
         project
-        and features.has("projects:similarity-embeddings-grouping", project)
+        and (
+            features.has("projects:similarity-embeddings-grouping", project)
+            or (
+                project.get_option("sentry:similarity_backfill_completed")
+                and randint(1, 100) <= options.get("similarity.delete_task_EA_rollout_percentage")
```
Comment (mrduncan) [inline src/sentry/tasks/delete_seer_grouping_records.py:64]: Just to confirm, we don't want any stickyness on the project/org and it's ok if it's a randomly distributed percentage

**Review by mrduncan — APPROVED** [review]

File: src/sentry/tasks/delete_seer_grouping_records.py (hunk 2 above)
Comment (jangjodi) [reply src/sentry/tasks/delete_seer_grouping_records.py:64]: Correct, we just want to make sure our task and seer can handle the load slowly. I don't think this slower rollout is strictly necessary, but wanted to be cautious.

**Pull request merged**